sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ItemsPartitionReader.scala

```diff
@@ -89,16 +89,24 @@ private case class ItemsPartitionReader
 
   private lazy val iterator = new TransientIOErrorsRetryingIterator(
     continuationToken => {
+
       if (!Strings.isNullOrWhiteSpace(continuationToken)) {
         ModelBridgeInternal.setQueryRequestOptionsContinuationTokenAndMaxItemCount(
-          queryOptions, continuationToken, readConfig.maxItemCount)
+          queryOptions, continuationToken,  readConfig.maxItemCount)
       } else {
         // scalastyle:off null
         ModelBridgeInternal.setQueryRequestOptionsContinuationTokenAndMaxItemCount(
           queryOptions, null, readConfig.maxItemCount)
         // scalastyle:on null
       }
 
+      queryOptions.setMaxBufferedItemCount(
+        math.min(
+          readConfig.maxItemCount * readConfig.prefetchBufferSize.toLong,
```

Comment:
nit: Stupid question but why convert it to long to compare that against an int?

Reply:
because maxItemCount and prefetchBufferSize are both INts
To prevent overflow (and becoming negative etc.)

Reply:
Several Spark 2 customers use Int32. MaxValue as maxItemCount to it could get to overflows easily

Reply:
`readConfig.prefetchBufferSize` and `readConfig.maxItemCount` are both ints. I guess there is nothing wrong with it operationally.

Reply:
Makes sense now? I thought it was clear - shoud I add a comment?

Reply:
All good, was missing the overflow scenario

Reply:
Added the comment - so we should be good now.